controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -221,52 +224,53 @@ func (r *StorageClassRequestReconciler) initPhase(storageProfile *v1.StorageProf
+		r.cephRadosNamespace.Namespace = r.OperatorNamespace
 		r.cephBlockPool = &rookCephv1.CephBlockPool{}
+		r.cephBlockPool.Name = profileName
 		r.cephBlockPool.Namespace = r.OperatorNamespace
-		for _, res := range r.StorageClassRequest.Status.CephResources {
-			if res.Kind == "CephBlockPool" {
-				r.cephBlockPool.Name = res.Name
-				break
-			}
+
+		// check if a CephBlockPoolRadosNamespace resource exists for the desired storageconsumer and storageprofile.
+		cephRadosNamespaceList := &rookCephv1.CephBlockPoolRadosNamespaceList{}
+		err := r.list(
+			cephRadosNamespaceList,
+			client.InNamespace(r.OperatorNamespace),
+			client.MatchingFields{util.OwnerUIDIndexName: string(r.StorageClassRequest.UID)})
+		if err != nil {
+			return err
 		}
 
-		// check if a cephblockpool resource exists for the desired storageconsumer and storageprofile.
-		if r.cephBlockPool.Name == "" {
-			cephBlockPoolList := &rookCephv1.CephBlockPoolList{}
-			listOptions := &client.MatchingLabels{
-				controllers.StorageConsumerNameLabel: r.storageConsumer.Name,
-				controllers.StorageProfileSpecLabel:  storageProfile.GetSpecHash(),
-			}
-			if err := r.list(cephBlockPoolList, client.InNamespace(r.OperatorNamespace), listOptions); err != nil {
+		// if we found no CephBlockPoolRadosNamespaces, generate a new name
+		// if we found only one CephBlockPoolRadosNamespace with our query, we're good
+		// if we found more than one CephBlockPoolRadosNamespace, we can't determine which one to select, so error out
+		rnsItemsLen := len(cephRadosNamespaceList.Items)
+		if rnsItemsLen == 0 {
+			err = r.get(r.cephBlockPool)
+			if err != nil {
 				return err
 			}
-
-			// if we found no CephBlockPools, generate a new name
-			// if we found only one CephBlockPool with our query, we're good
-			// if we found more than one CephBlockPool, we can't determine which one to select, so error out
-			cbpItemsLen := len(cephBlockPoolList.Items)
-			if cbpItemsLen == 0 {
-				cbpNewName := fmt.Sprintf("cephblockpool-%s-%s", r.storageConsumer.Name, generateUUID())
-				r.log.V(1).Info("no valid CephBlockPool found, creating new one", "CephBlockPool", cbpNewName)
-				r.cephBlockPool.Name = cbpNewName
-			} else if cbpItemsLen == 1 {
-				r.cephBlockPool.Name = cephBlockPoolList.Items[0].GetName()
-				r.log.V(1).Info("valid CephBlockPool found", "CephBlockPool", r.cephBlockPool.Name)
-			} else {
-				return fmt.Errorf("invalid number of CephBlockPools for storage consumer %q and storage profile %q: found %d, expecting 0 or 1", r.storageConsumer.Name, storageProfile.Name, cbpItemsLen)
-			}
+			md5Sum := md5.Sum([]byte(r.StorageClassRequest.Name))
+			rnsNewName := fmt.Sprintf("cephradosnamespace-%s", hex.EncodeToString(md5Sum[:16]))
+			r.log.V(1).Info("no valid CephBlockPoolRadosNamespace found, creating new one", "CephBlockPoolRadosNamespace", rnsNewName)
+			r.cephRadosNamespace.Name = rnsNewName
+			r.cephRadosNamespace.Spec.BlockPoolName = profileName
```

Comment:
On second thought, the block pool name is part of the sepc and needs to be edited inside the `createOrUpdate` if not it might get overwritten by the `Get` operation.  
Let's remove this line
```suggestion
```